src/api/Nhs.Appointments.Persistance/BookingCosmosDocumentStore.cs

```diff
@@ -90,6 +92,23 @@ public async Task<bool> UpdateStatus(string bookingReference, string status)
         return true;
     }
 
+    public async Task<bool> ConfirmProvisional(string bookingReference, IEnumerable<ContactItem> contactDetails)
+    {
+        var bookingIndexDocument = await _indexStore.GetDocument<BookingIndexDocument>(bookingReference);
+        if (bookingIndexDocument == null)
+        {
+            return false;
+        }
+
+        var updateStatusPatch = PatchOperation.Replace("/provisional", false);
+        await _bookingStore.PatchDocument(bookingIndexDocument.Site, bookingReference, updateStatusPatch);
+        await _indexStore.PatchDocument("booking_index", bookingReference, updateStatusPatch);
+
+        var addContactDetailsPath = PatchOperation.Add("/contactDetails", contactDetails);
+        await _bookingStore.PatchDocument(bookingIndexDocument.Site, bookingReference, addContactDetailsPath);
```

Comment:
Patch document can accept multiple Patch operation and will apply them atomically, so we can do this for the bookingStore update